nobodywho/core/src/chat.rs

```diff
@@ -50,6 +50,33 @@ pub struct ChatHandle {
     should_stop: Arc<AtomicBool>,
 }
 
+///
+/// Configuration for chat sessions.
+///
+/// This struct groups all the settings needed to initialize a chat worker.
+/// Use [`ChatBuilder`] for a more ergonomic way to configure these settings.
+pub struct ChatConfig {
+    /// Available tools for the model to use.
+    pub tools: Vec<Tool>,
+    /// Context window size.
+    pub n_ctx: u32,
+    /// System prompt for the chat session.
+    pub system_prompt: String,
+    /// Whether to enable thinking mode during inference.
+    pub enable_thinking: bool,
+}
+
+impl Default for ChatConfig {
+    fn default() -> Self {
+        Self {
+            n_ctx: 2048,
```

Comment:
I think 4096 would be better as a default context size, if for no other reason than to match the defaults set in e.g. godot/src/lib.rs (around line 158). 

Actually, now that we have defaults for this type of thing in core, it would probably be good to initialize the configurable godot node with those defaults (again: godot/src/lib.rs)